Sources/SPMBuildCore/BinaryTarget+Extensions.swift

```diff
@@ -69,9 +69,11 @@ extension BinaryTarget {
         // Construct an ExecutableInfo for each matching variant.
         return try executables.flatMap { entry in
             // FIXME: this filter needs to become more sophisticated
-            try entry.value.variants.filter {
-                return $0.supportedTriples.contains(versionLessTriple)
-            }.map{
+            let filtered = entry.value.variants.filter { $0.supportedTriples.contains(versionLessTriple) }
+            if filtered.isEmpty {
+                observabilityScope?.emit(warning: "binary '\(self.name)' not on supported host/platform (\(versionLessTriple.tripleString))")
+            }
+            return try filtered.map{
```

Comment:
I think we need a bigger refactoring here to emit these diagnostics effectively, since right now we would warn also for any executable that doesn't end up being used. 

We would probably need to change `ExecutableInfo` and `PluginAccessibleTool` to include supported platform info and thread that all the way down to the plugin context, so that we can have a more targeted error from `tool(named:)` since we won't know if a tool is being used until then.

Reply:
makes sense 👍